#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/road/protos/RoadLine.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/road/protos/StraightRoadSegment.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/vehicles/protos/toyota/ToyotaPrius.proto"

WorldInfo {
  info [
    "Autonomous Vehicle Simulation"
    "The simple controller example uses an on-board camera to follow the yellow road lines and a SICK sensor to avoid the obstacles."
    "The control of the vehicle is done using the driver library."
    "The vehicle based on the Car PROTO is modelled with realistic physics properties: motor torques, body mass, friction forces, suspensions, etc."
  ]
  title "City"
  ERP 0.6
  basicTimeStep 10
  lineScale 1
  contactProperties [
    ContactProperties {
      coulombFriction [
        8
      ]
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    ContactProperties {
      material2 "CitroenCZeroWheels"
      coulombFriction [
        8
      ]
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    ContactProperties {
      material2 "ToyotaPriusWheels"
      coulombFriction [
        8
      ]
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    ContactProperties {
      material2 "LincolnMKZWheels"
      coulombFriction [
        8
      ]
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    ContactProperties {
      material2 "RangeRoverSportSVRWheels"
      coulombFriction [
        8
      ]
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    ContactProperties {
      material2 "TruckWheels"
      coulombFriction [
        8
      ]
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
  ]
}
Viewpoint {
  orientation 0.13123143069783055 0.024202245667947882 -0.9910562864447351 2.7800125655321826
  position 530.5974272294591 18.34363990840633 14.258537959154395
  near 1
  followType "None"
  followSmoothness 0
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Fog {
  color 0.38 0.35 0.32
  visibilityRange 1000
}
DEF GROUND Solid {
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0.8
        baseColorMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/vehicles/worlds/textures/ground.jpg"
          ]
        }
        roughness 0.5
        metalness 0
        textureTransform TextureTransform {
          scale 500 500
        }
      }
      geometry DEF GROUND_PLANE Plane {
        size 3000 2000
      }
    }
  ]
  boundingObject USE GROUND_PLANE
  locked TRUE
}
DEF ROAD_SEGMENT StraightRoadSegment {
  translation 0 0 0.02
  name "road(3)"
  id "3"
  startJunction "23"
  endJunction "22"
  width 14.8
  numberOfLanes 4
  numberOfForwardLanes 2
  lines [
    RoadLine {
    }
    RoadLine {
      color 0.85 0.75 0.3
      type "double"
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  rightBarrier TRUE
  leftBarrier TRUE
  length 2000
}
DEF SCONE Robot {
  translation 500 -5.6 0.35
  children [
    Emitter {
      name "cone_emitter"
    }
    Gyro {
    }
    GPS {
    }
    DEF ACCELEROMETER Accelerometer {
    }
    Solid {
      translation 0 0 -0.33
      children [
        DEF BASE-SHAPE Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Box {
            size 0.38 0.38 0.05
          }
        }
      ]
      name "solid(1)"
    }
    Solid {
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0.952941 0.32549 0.0117647
            }
          }
          geometry Cone {
            bottomRadius 0.15
            height 0.71
            subdivision 50
          }
        }
      ]
    }
  ]
  name "SCONE"
  boundingObject Cylinder {
    height 0.71
    radius 0.15
  }
  physics Physics {
    mass 3.63
    damping Damping {
    }
  }
  controller "traffic_cone_controller"
}
DEF SUPERVISOR Robot {
  children [
    Emitter {
      name "supervisor_emitter"
      channel 1
    }
    Receiver {
      name "supervisor_receiver"
    }
  ]
  controller "workzone_supervisor_controller"
  supervisor TRUE
}
DEF TOYOTA ToyotaPrius {
  translation 10 -5.6 0.35
  controller "custom_car_controller"
  sensorsSlotFront [
    Receiver {
      name "car_receiver"
      channel 1
    }
  ]
}
